### PR TITLE
Add a command to batch destroy instances.

### PR DESCRIPTION
**Pull request by tai271828**
The current command gce-delete-instance could only delete one instance
at a time. This command uses gce-delete-instance in a for loop to
batch destroy instances. This command is useful when creating a test
cluster which will be destoried very soon to avoid costly GCE bill.

**Test Case**
1. Create several instances. `gstart instance01; gstart instance02; gstart instance03; gstart instance04; gstart instance05; gstart instance06`
2. `gce-delete-instance`[1]
3. `gce-delete-instance instance01`[2]
4. `gce-delete-instance instance01 -q`[3]
5. `gce-delete-instance -a`[4]
6. `gce-delete-instance -a -q`[5]

**Expected Result**

[1] This command should does nothing but complain `No instance is given.`
[2] This command should prompt to delete `instance01`
[3] This command should delete `instance01` without prompt.
[4] This command should try to delete all instances with prompt.
[5] This command should try to delete all instances without prompt.

### CODE REVIEW

File: bin/gce-delete-instances
Comment (yungyuc) [inline bin/gce-delete-instances]: Please merge this script with `gce-delete-instance`.  Two scripts differing in the trailing character may cause confusion.

`gce-delete-instace --all` may work?

Comment (yungyuc) [inline bin/gce-delete-instances]: Please merge this script with `gce-delete-instance`.  Two scripts differing in the trailing character may cause confusion.

`gce-delete-instace --all` may work?

**Review by yungyuc — CHANGES_REQUESTED** [review]

File: bin/gce-delete-instance
Comment (tai271828) [inline bin/gce-delete-instance:20]: This is an extra shift to parse argument because `quite` option should has no argument.

File: bin/gce-delete-instances
Comment (tai271828) [reply bin/gce-delete-instances]: `gce-delete-instace --all` is good to me. Actually this is the first idea came to my mind. I did not propose this version because this needs to largely refactor the code. Besides, two-script implementation is more readable (to me) but may cause confusion to users as what you mentioned. I will create another PR soon for the `gce-delete-instace --all` and let's compare which one is better.

Comment (yungyuc) [reply bin/gce-delete-instances]: It's OK to update this PR.  I don't think having two scripts is a good idea.

File: bin/gce-delete-instance
Comment (tai271828) [inline bin/gce-delete-instance]: The original `gce-delete-instance` is wrapped in the function `delete-instance`.

File: bin/gce-delete-instance
Comment (tai271828) [inline bin/gce-delete-instance]: Call the `delete-instance` in a for loop to batch delete instances.

File: bin/gce-delete-instances
Comment (tai271828) [reply bin/gce-delete-instances]: PR is updated. Please review it again. Thanks.

**Review by tai271828 — COMMENTED** [review]
PR is updated. We could use `gce-delete-instance` as usual, and use one more option `gce-delete-instance --all` as well. Please review it again. Thanks.

File: bin/gce-delete-instance
Comment (yungyuc) [inline bin/gce-delete-instance]: The usage string should be combined with the `--all` case.  I would need you to help me fix the typo (`usage` instead of `uage`).

File: bin/gce-delete-instance
Comment (yungyuc) [inline bin/gce-delete-instance]: Typo: `instnaces`.

It could be clearer to use `[instance1 [instance2 [...]]]`.

If you prefer to use the current form, `ex.` should be `e.g.`.

File: bin/gce-delete-instance
Comment (yungyuc) [inline bin/gce-delete-instance]: This argument assignment should be moved outside this function.

File: bin/gce-delete-instance
Comment (yungyuc) [inline bin/gce-delete-instance]: We still should stop execution when `instname` is void, but we don't need to print usage.

File: bin/gce-delete-instance
Comment (yungyuc) [inline bin/gce-delete-instance]: My typo: `quiet`, not `quite`.

**Review by yungyuc — CHANGES_REQUESTED** [review]

File: bin/gce-delete-instance
Comment (tai271828) [reply bin/gce-delete-instance]: `[instance1 [instance2 [...]]]` is better. Thanks!

File: bin/gce-delete-instance
Comment (tai271828) [reply bin/gce-delete-instance]: Do you mean to use argument assignment in a function when writing a shell script is not encouraged?

File: bin/gce-delete-instance
Comment (tai271828) [reply bin/gce-delete-instance]: Fixed in the latest commit. https://github.com/solvcon/solvcon-gce/pull/19/commits/f338780584502392e3ef7e3982244c7e47f2f199

File: bin/gce-delete-instance
Comment (tai271828) [reply bin/gce-delete-instance]: Fixed in the latest commit. https://github.com/solvcon/solvcon-gce/pull/19/commits/f338780584502392e3ef7e3982244c7e47f2f199

Use `No instance is given.` and exit instead.

File: bin/gce-delete-instance
Comment (tai271828) [reply bin/gce-delete-instance]: Fixed in the latest commit. https://github.com/solvcon/solvcon-gce/pull/19/commits/f338780584502392e3ef7e3982244c7e47f2f199

**Review by tai271828 — COMMENTED** [review]
PR is updated according to the discussion thread. Please review it. If both of us think the PR is good to us, I will squash the commit.

**Review by tai271828 — COMMENTED** [review]
Test cases in the PR description has been updated as well for the latest commit. You could play around with it according to the test case instruction in the description.

File: bin/gce-delete-instance
Comment (yungyuc) [inline bin/gce-delete-instance:14]: Thank your for fixing my typo :)

**Review by yungyuc — APPROVED** [review]
Looks great!

**Comment by yungyuc** [comment]
I wait for you to squash the changes before merge.

**Comment by tai271828** [comment]
The branch has been updated to squash the commits, and I have updated this PR with that branch. It is ready to merge. Thanks.

**Comment by yungyuc** [comment]
Thanks for the PR.  Merged.

**Pull request merged**
